keyboards/yushakobo/primer16/keymaps/via/keymap.c

```diff
@@ -0,0 +1,138 @@
+                 KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO
+  //|-----------------------------------|
+  ),
+  [2] = LAYOUT(
+           //,-----------------|
+                 KC_NO,   KC_NO,
+           //|--------+--------|
+                 KC_NO,   KC_NO,
+           //,-----------------|
+  //,-----------------------------------|
+                 KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO
+  //|-----------------------------------|
+  ),
+  [3] = LAYOUT(
+           //,-----------------|
+                 KC_NO,   KC_NO,
+           //|--------+--------|
+                 KC_NO,   KC_NO,
+           //,-----------------|
+  //,-----------------------------------|
+                 KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,
+  //|--------+--------+--------+--------|
+        KC_NO,   KC_NO,   KC_NO,   KC_NO
+  //|-----------------------------------|
+  )
+};
+
+//A description for expressing the layer position in LED mode.
+layer_state_t layer_state_set_user(layer_state_t state) {
+#ifdef RGBLIGHT_ENABLE
+    switch (get_highest_layer(state)) {
+    case 1:
+      rgblight_sethsv_at(HSV_BLUE, 0);
+      break;
+    case 2:
+      rgblight_sethsv_at(HSV_RED, 0);
+      break;
+    case 3:
+      rgblight_sethsv_at(HSV_GREEN, 0);
+      break;
+    default: //  for any other layers, or the default layer
+      rgblight_sethsv_at( 0, 0, 0, 0);
+      break;
+    }
+    rgblight_set_effect_range( 1, 5);
+#endif
+return state;
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    keypos_t key;
+    if (index == 0) {
+        if (clockwise) {
+            key.row = 5;
+            key.col = 1;
+        } else {
+            key.row = 6;
+            key.col = 1;
+        }
+    } else if (index == 1) {
+        if (clockwise) {
+            key.row = 5;
+            key.col = 2;
+        } else {
+            key.row = 6;
+            key.col = 2;
+        }
+    }
+    action_exec((keyevent_t){.key = key, .pressed = true, .time = (timer_read() | 1)});
+    action_exec((keyevent_t){.key = key, .pressed = false, .time = (timer_read() | 1)});
+    return true;
+}
```

Comment:
This method is no longer being accepted into the repo, and https://docs.qmk.fm/#/feature_encoders?id=encoder-map should be used instead.

Reply:
I am using this write method to remap rotary encoder rotations in VIA.
Is it possible to do the same with this ENCODER_MAP_ENABLE?
If not possible, I would prefer to leave it as is.

Reply:
https://github.com/qmk/qmk_firmware/pull/17734